src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -166,10 +166,9 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
             if (resp.isAllowedAccess) {
               if (resp.isUsed) {
                 // The logged in user matches the registration key; redirect to the logged in URL
-
                 this.navigationService.navigateByURLWithParamEncoding(
                     this.router, `/web/${this.entityType}/sessions/submission`,
-                    { courseid: this.courseId, fsname: this.feedbackSessionName });
+                    { courseid: this.courseId, fsname: this.feedbackSessionName, entitytype: this.entityType });
```

Comment:
I added this query param for the case where a session not found unhappy path is hit when a logged-in instructor clicks on its associated feedback session link, the instructor is redirected to the instructor home page after closing/dismissing the modal. Without this query param, the instructor is redirected to the student home page instead. This is because `this.entityType` resets to `student` after redirection to instructor submission page.

Reply:
That is indeed something to be corrected, but this shouldn't be the way to do it.
A few lines above there should be something like:
```ts
tap((data: any) => {
  this.intent = data.intent;
}),
```
What should be done is modifying `this.entityType` in this block if the intent matches.